### PR TITLE
feat: new artifact for sdk-platform-java configs. 

### PR DESCRIPTION
**Pull request by mpeddada1**
This artifact is meant to mirror java-shared-config with the java-shared-dependencies version to then be consumed by downstream libraries (including the monorepo and handwritten libraries). The purpose of this artifact is to have a single update PR in the downstream libs to update shared-config and shared-dependencies instead of two. 

Name of the artifact: `sdk-platform-java-config`

### CODE REVIEW

**Review by burkedavison — COMMENTED** [review]
LGTM, let's get Tomo's approval as well since this is creating a new artifact.

**Review by burkedavison — APPROVED** [review]

**Comment by mpeddada1** [comment]
@suztomo Could we get your final sign off on this PR?

**Review by suztomo — APPROVED** [review]

**Review by burkedavison — APPROVED** [review]

**Review by suztomo — APPROVED** [review]

**Comment by suztomo** [comment]
10 mins after merging this, ensure the Release Please pull request https://github.com/googleapis/sdk-platform-java/pull/2290 updates appropriate lines.

**Comment by mpeddada1** [comment]
Will do! Thanks

File: sdk-platform-java-config/pom.xml
Comment (suztomo) [inline sdk-platform-java-config/pom.xml:20]: (I see it's merged) What's the point of `<checkstyle.skip>true</checkstyle.skip>`?

Comment (suztomo) [reply sdk-platform-java-config/pom.xml:20]: https://github.com/googleapis/sdk-platform-java/pull/2335

**Pull request merged**
